haiku-kata-solutions/src/main/java/org/eclipse/collections/haikukata/TextProcessorEC.java

```diff
@@ -77,18 +77,22 @@ public Triple<CharBag, CharBag, CharSet> duplicatesAndUnique()
 
     public CharCharPair topVowelAndConsonant()
     {
-        // TODO: Find all of the alphabetic letters from this.getHaikuAsCharAdapter(), convert them to lowercase,
-        // TODO: put them in a bag and then get the top 26 occurrences
-        // Hint: Look at select, collectChar, toBag, and topOccurrences
-        // Bonus: See if the same solution will work using asLazy
-        MutableList<CharIntPair> charIntPairs = null;
-
-        // TODO: Find the top vowel
-        // Hint: Use the detect method on MutableList with the isVowel method below to find the top vowel char value
-        char topVowel = 'a';
-        // TODO: Find the top consonant
-        // Hint: Use the detect method on MutableList with the isVowel method below to find the top consonant char value
-        char topConsonant = 'b';
+        private static final char DEFAULT_CHAR = ' ';
+        // Collecting all the alphabetic letters from getHaikuAsCharAdapter(), converting them to lowercase,
+        // putting them in a bag and then getting the top 26 occurrences.
+        MutableList<CharIntPair> charIntPairs = this.getHaikuAsCharAdapter()
+                                                     .select(Character::isAlphabetic)
+                                                     .collectChar(Character::toLowerCase)
+                                                     .toBag()
+                                                     .topOccurrences(26);
+
+        // Finding the top vowel.
+        CharIntPair topVowelPair = charIntPairs.detect(pair -> isVowel(pair.getOne()));
+        char topVowel = topVowelPair == null ? ' ' : topVowelPair.getOne();  // Default to space if no vowel is found.
```

Comment:
We can use DEFAULT_CHAR here as well

Reply:
oops yes sorry
Rachana I not quite sure about how to squash the comments

Reply:
Can you please help